aws_lambda_powertools/metrics/functions.py

```diff
@@ -69,3 +72,66 @@ def extract_cloudwatch_metric_unit_value(metric_units: List, metric_valid_option
         unit = unit.value
 
     return unit
+
+
+def validate_emf_timestamp(timestamp: int | datetime) -> bool:
+    """
+    Validates a given timestamp based on CloudWatch Timestamp guidelines.
+
+    Timestamp must meet CloudWatch requirements, otherwise an InvalidTimestampError will be raised.
+    See [Timestamps](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#about_timestamp)
+    for valid values.
+
+    Parameters:
+    ----------
+    timestamp: int | datetime
+        Datetime object or epoch time in milliseconds representing the timestamp to validate.
+
+    Returns
+    -------
+    bool
+        Valid or not timestamp values
+    """
+
+    if not isinstance(timestamp, (int, datetime)):
+        return False
+
+    if isinstance(timestamp, datetime):
+        # Converting timestamp to epoch time in milliseconds
+        timestamp = int(timestamp.timestamp() * 1000)
+
+    # Consider current timezone when working with date and time
+    current_timezone = datetime.now().astimezone().tzinfo
```

Comment:
I'm thinking that this line could be moved outside the function, as it will always be the same during a Lambda session. I've done a quick benchmark and calling `astimezone()` twice on each call is by far the most expensive part of the operation. What do you think?

Reply:
What if for some reason the customer sets a different timezone at runtime? For example, in the case of a multi-tenant application.

Do you think it’d be less expensive to move the function outside and initialize the value, but then within the function check if the `TZ` value has changed, and only thenupdate the `current_timezone` value?

Reply:
Ah mutable global TZ settings, what can possibly go wrong? :D 

Yes, you're right, feel free to ignore my request @leandrodamascena!

Reply:
>Ah mutable global TZ settings, what can possibly go wrong? :D

"Beautiful and scary things in Python - Part 1" haha.

Yeah, I agree with @dreamorosi! Let's keep it as is.

Reply:
I think if we cache both `current_timezone` (which is the expensive part) and `TZ` outside of the function, we could shave some ms by checking `TZ` against `os.getEnv(‘TZ’)` and updating the value only if it changed.

Although that would make the implementation a bit more complex..